src/Platform/Microsoft.Testing.Platform.MSBuild/Tasks/DotnetMuxerLocator.cs

```diff
@@ -401,24 +401,21 @@ public bool TryGetDotnetPathByArchitecture(
         try
         {
             using var headerReader = new FileStream(path, FileMode.Open, FileAccess.Read);
-            byte[] magicBytes = new byte[4];
+
+            // skip magicBytes by moving forward 4 bytes
+            headerReader.Position += 4;
```

Comment:
Maybe it's better check it and throw if it's not what we expect, we should fix it also here https://github.com/microsoft/vstest/blob/ece7f1653908d4eed603ae4cf0122a54d6df6a74/src/Microsoft.TestPlatform.CoreUtilities/Helpers/DotnetHostHelper.cs#L410

@nohwnd I recall you wrote this part do you agree?

Reply:
we ended up using the magic bytes. https://github.com/microsoft/testfx/issues/6032